plugins/modules/s3_metrics_configuration.py

```diff
@@ -0,0 +1,246 @@
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_tag_list
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+
+
+def _create_metrics_configuration(mc_id, filter_prefix, filter_tags):
+    payload = {
+        'Id': mc_id
+    }
+    if len(filter_tags) == 1 and filter_prefix is None:
+        payload['Filter'] = {
+            'Tag': ansible_dict_to_boto3_tag_list(filter_tags)[0]
+        }
+    elif len(filter_tags) == 0 and filter_prefix is not None:
+        payload['Filter'] = {
+            'Prefix': filter_prefix
+        }
+    elif len(filter_tags) > 0:
+        payload['Filter'] = {
+            'And': {
+                'Tags': ansible_dict_to_boto3_tag_list(filter_tags)
+            }
+        }
+        if filter_prefix is not None:
+            payload['Filter']['And']['Prefix'] = filter_prefix
+
+    return payload
+
+
+def _compare_metrics_configuration(metrics_configuration, mc_id, filter_prefix, filter_tags):
+    payload = metrics_configuration['MetricsConfiguration']
+    parsed = {'mc_id': payload.get('Id')}
+    if payload.get('Filter', {}).get('Prefix') is not None:
+        parsed['filter_prefix'] = payload['Filter']['Prefix']
+    if payload.get('Filter', {}).get('Tag') is not None:
+        parsed['filter_tags'] = boto3_tag_list_to_ansible_dict([payload['Filter']['Tag']])
+    if payload.get('Filter', {}).get('And') is not None:
+        parsed['filter_tags'] = boto3_tag_list_to_ansible_dict(payload['Filter']['And'].get('Tags', []))
+        if payload['Filter']['And'].get('Prefix') is not None:
+            parsed['filter_prefix'] = payload['Filter']['And']['Prefix']
+
+    if parsed.get('mc_id') != mc_id:
+        return False
+    if parsed.get('filter_prefix') != filter_prefix:
+        return False
+    if parsed.get('filter_tags', {}) != filter_tags:
+        return False
+    return True
+
+
+def create_or_update_metrics_configuration(client, module):
+    changed = False
+    bucket_name = module.params.get('bucket_name')
+    mc_id = module.params.get('id')
+    filter_prefix = module.params.get('filter_prefix')
+    filter_tags = module.params.get('filter_tags')
+
+    try:
+        metrics_configuration = client.get_bucket_metrics_configuration(Bucket=bucket_name, Id=mc_id)
+    except is_boto3_error_code('NoSuchConfiguration'):
+        metrics_configuration = None
+    except ClientError as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(e, msg="Failed to get bucket metrics configuration")
+
+    if metrics_configuration:
```

Comment:
There is already `_create_metrics_configuration()` which converts mc_id/filter_prefix/filter_tags into a dict in the same format as metrics_configuration is there a reason to compare them in a different format?

I think it should also be possible to simplify this logic a little:

```
new_configuration = _create_metrics_configuration(mc_id, filter_prefix, filter_tags)

if metrics_configuration:
    if metrics_configuration == new_configuration
        module.exit_json(changed=False)

if module.check_mode:
    module.exit_json(changed=True)

# Create and update both use the same call
<set_configuration>

module.exit_json(changed=True)
```